hl7_pims_transformer/hl7_pims_transformer/mappers/pid_mapper.py

```diff
@@ -8,14 +8,36 @@ def map_pid(original_hl7_message: Message, new_message: Message) -> None:
     if not original_pid:
         return  # No PID segment
 
+    # trigger event for A28 specific logic
+    trigger_event = get_hl7_field_value(new_message, "msh.msh_9.msg_2")
```

Comment:
so we're relying on the MSH mapper being run first to have converted the original message type to A28, A31 or A40 in the new message, would be good to add a comment explicitly saying that as it's really easy to miss and might be a sneaky source of bugs later should the mappers not be run in this order.

Reply:
Alternatively I would use the original message and knowing from the msh_mapper that
`original_message_type_trigger_event = original_msh.msh_9.msg_2` is `A04` and that maps to `A28`, use `A04` as a check value instead of relying on the mapping order, so that we break that dependency between mappers.